src/System.IO.FileSystem.Watcher.Polling/System/IO/PathToFileStateHashtable.cs

```diff
@@ -121,41 +101,21 @@ private int NextCandidateBucket(int bucket)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private unsafe bool Equal(FullPath fullPath, string directory, char* file)
+        private unsafe bool Equal(FullPath fullPath, string directory, ReadOnlySpan<char> file)
         {
-
             if (!String.Equals(fullPath.Directory, directory, StringComparison.Ordinal))
             {
                 return false;
             }
 
-            int index;
-            for (index = 0; index < fullPath.File.Length; index++)
-            {
-                if (file[index] == 0) return false;
-                if (file[index] != fullPath.File[index]) return false;
-            }
-            if (file[index] != 0) return false;
-            return true;
-
-        }
-
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private unsafe bool Equal(FullPath fullPath, string directory, string file)
-        {
-            if (!String.Equals(fullPath.Directory, directory, StringComparison.Ordinal))
-            {
-                return false;
-            }
-
-            if (!String.Equals(fullPath.File, file, StringComparison.Ordinal))
+            if (!MemoryExtensions.Equals(fullPath.File, file, StringComparison.Ordinal))
```

Comment:
Nit- use the extension syntax. (`fullPath.File.Equals(...`)

Reply:
Other way around, actually. :) `file.Equals(fullPath.File...`  (so you don't have to explicitly cast)